lib/Sema/CSSolver.cpp

```diff
@@ -922,8 +922,9 @@ void ConstraintSystem::Candidate::applySolutions(
     if (OSR->getDecls().size() == choices.size()) continue;
 
     // Update the expression with the reduced domain.
-    MutableArrayRef<ValueDecl *> decls
-      = TC.Context.AllocateUninitialized<ValueDecl *>(choices.size());
+    MutableArrayRef<ValueDecl *> decls =
+        TC.Context.AllocateUninitialized<ValueDecl *>(choices.size(),
+                                                      Allocator);
```

Comment:
I think I would just skip the whole change for ASTContext and use the BumpPtrAllocator directly. You should be able to do this:

```c++
MutableArrayRef<ValueDecl *> decls(Allocator.Allocate<ValueDecl *>(choices.size()),
                                   choices.size());
```